src/test/java/com/puppycrawl/tools/checkstyle/BaseCheckTestSupport.java

```diff
@@ -165,17 +169,45 @@ protected static void verifyAst(String expectedTextPrintFileName, String actualJ
      * @throws Exception if exception occurs during verification.
      */
     protected static void verifyAst(String expectedTextPrintFileName, String actualJavaFileName,
-            boolean withComments) throws Exception {
-        final String expectedContents = new String(Files.readAllBytes(
-            Paths.get(expectedTextPrintFileName)), StandardCharsets.UTF_8)
-            .replaceAll("\\\\r\\\\n", "\\\\n");
+                                    boolean withComments) throws Exception {
+        final String expectedContents = readFile(expectedTextPrintFileName);
+
         final String actualContents = AstTreeStringPrinter.printFileAst(
-                new File(actualJavaFileName), withComments).replaceAll("\\\\r\\\\n", "\\\\n");
+                new File(actualJavaFileName), withComments).replaceAll(CLRF_REGEX, LF_REGEX);
+
+        assertEquals("Generated AST from Java file should match pre-defined AST", expectedContents,
+                actualContents);
+    }
+
+    /**
+     * Verifies the Javadoc Ast generated for the supplied java file against the supplied text file.
+     * @param expectedTextPrintFileName expected text ast tree representation.
+     * @param actualJavaFileName actual text ast tree representation.
+     * @throws Exception if exception occurs during verification.
+     */
+    protected static void verifyJavadocAst(String expectedTextPrintFileName,
```

Comment:
I was also referring to class names.
https://github.com/checkstyle/checkstyle/blob/4a087c020c2b97130e94f9292b94cb4bc1f6ed93/src/main/java/com/puppycrawl/tools/checkstyle/api/DetailAST.java
https://github.com/checkstyle/checkstyle/blob/b4a337a599e853cdc44b19a76da28fcfab2956cd/src/main/java/com/puppycrawl/tools/checkstyle/api/DetailNode.java

Naming is because old ANTLR built AST nodes, as we extend `CommonASTWithHiddenTokens`, but ANTLR4 doesn't support AST nodes of old ANTLR, hence why it isn't called `AST`.